Use Documentation Tags for Maven and Gradle in Getting Started

harpreets789 opened the pull request:
This PR consolidates the separate XML and Gradle code snippets into joint tabs within the getting-spring-security.adoc file
Issue gh-16228

jzheaux commented on docs/modules/ROOT/pages/getting-spring-security.adoc:
```diff
@@ -15,19 +15,22 @@ Typically, these are done to provide improved security to match modern security
 
 
 [[maven]]
```
Please add `[[getting-gradle]]` so that bookmarks to this anchor still work

jzheaux commented on docs/modules/ROOT/pages/getting-spring-security.adoc:
```diff
@@ -15,19 +15,22 @@ Typically, these are done to provide improved security to match modern security
 
 [[maven]]
-== Usage with Maven
+== Usage with Maven and Gradle
```
I think `Usage` is just fine at this point.

jzheaux commented on docs/modules/ROOT/pages/getting-spring-security.adoc:
```diff
@@ -15,19 +15,22 @@ Typically, these are done to provide improved security to match modern security
 [[getting-maven-boot]]
-=== Spring Boot with Maven
+[[getting-gradle-boot]]
+=== Spring Boot with Maven and Gradle
```
Since you've already stated that you'll be talking about both, it's okay to simply say `Spring Boot`.

jzheaux commented on docs/modules/ROOT/pages/getting-spring-security.adoc:
```diff
@@ -40,9 +43,25 @@ Alternatively, you can manually add the starter, as the following example shows:
+
 Since Spring Boot provides a Maven BOM to manage dependency versions, you do not need to specify a version.
-If you wish to override the Spring Security version, you can do so by providing a Maven property:
+If you wish to override the Spring Security version, you can do so by providing a Maven property or declaring it as a Gradle property, as shown below:
```
Let's see if we can keep it still more generic to simplify the wording. Maybe:

```
If you wish to override the Spring Security version, you can do so with a build property as shown below:
```

jzheaux commented on docs/modules/ROOT/pages/getting-spring-security.adoc:
```diff
@@ -52,10 +71,24 @@ If you wish to override the Spring Security version, you can do so by providing
 Since Spring Security makes breaking changes only in major releases, you can safely use a newer version of Spring Security with Spring Boot.
 However, at times, you may need to update the version of Spring Framework as well.
-You can do so by adding a Maven property:
+You can do so by adding a Maven or Gradle property:
```
Let's consider the same simplification here:

```
You can do so by adding a build property like so:
```

jzheaux commented on docs/modules/ROOT/pages/getting-spring-security.adoc:
```diff
@@ -65,13 +98,27 @@ You can do so by adding a Maven property:
+=== Standalone Usage (Without Spring Boot)
 
-When you use Spring Security without Spring Boot, the preferred way is to use Spring Security's BOM to ensure that a consistent version of Spring Security is used throughout the entire project. The following example shows how to do so:
+When you use Spring Security without Spring Boot, the preferred way is to use Spring Security's BOM to ensure that a consistent version of Spring Security is used throughout the entire project.  For Gradle, you can do so by using the https://github.com/spring-gradle-plugins/dependency-management-plugin[Dependency Management Plugin]:
```
I'd recommend leaving this link to the plugin until after the snippet and also inside a `TIP`:

```
[TIP]
Spring provides a `Dependency Management Plugin`(link) for Gradle
```

I'd recommend doing the same with the same link later on in the article.

jzheaux commented on docs/modules/ROOT/pages/getting-spring-security.adoc:
```diff
@@ -138,12 +243,24 @@ This approach uses Maven's "`bill of materials`" (BOM) concept and is only avail
-=== Maven Repositories
-All GA releases (that is, versions ending in .RELEASE) are deployed to Maven Central, so you need not declare additional Maven repositories in your pom.
+[[maven-gradle-gearepositories]]
+=== Maven and Gradle Repositories
```
I think this should say as "Maven Repositories" since that's what they are in either case.

jzheaux commented on docs/modules/ROOT/pages/getting-spring-security.adoc:
```diff
@@ -138,12 +243,24 @@ This approach uses Maven's "`bill of materials`" (BOM) concept and is only avail
-All GA releases (that is, versions ending in .RELEASE) are deployed to Maven Central, so you need not declare additional Maven repositories in your pom.
+[[maven-gradle-gearepositories]]
+=== Maven and Gradle Repositories
+All GA releases (that is, versions ending in .RELEASE) are deployed to Maven Central, so you need not declare additional Maven repositories in your `pom.xml` or, for Gradle, using the `mavenCentral()` repository is sufficient for GA releases.
```
Please place the "For Gradle" in a separate sentence and separate paragraph. This makes the sentence shorter and continue to be about one thing.

harpreets789 commented:
Thank you for your thorough review and valuable suggestions. I appreciate the directions you've provided and will make the necessary changes accordingly.

harpreets789 commented:
> @jzheaux  I have addressed the PR review feedback and made the necessary updates to the "getting-spring-security" documentation. Please let me know if there are any additional changes required!

jzheaux commented:
Thanks, @harpreets789! This is now merged into `main`.
